tests/test_black.py

```diff
@@ -341,6 +341,20 @@ def test_string_quotes(self) -> None:
         black.assert_equivalent(source, not_normalized)
         black.assert_stable(source, not_normalized, mode=mode)
 
+    def test_skip_source_first_line(self) -> None:
```

Comment:
Add a test case where there is some valid code after the first line.

Also, what if the file doesn't use `\n` to separate lines?

Reply:
I'll add a valid code after the first line in the test file:

```python
This is not valid Python syntax
y = "This is valid syntax"
```

About the second concern, it is indeed valid. I solved it by using the `newline` variable that comes from `decode_bytes` since it's already available. But now several Blackd tests fail because the variable is not available when Blackd invokes `black.format_file_contents`.

https://github.com/psf/black/blob/07b68e2425d2caadc2edf8fe021f60a32c6f44a7/src/blackd/__init__.py#L132-L134

This seems related to a WIP at #3257 . Maybe a `newline` variable could be defined in Blackd before calling `black.format_file_contents`, using the same check that's implemented in the cited PR, something like:

```python
if req_str[req_str.find("\n") - 1] == "\r":
    newline = "\r\n"
else:
    newline = "\r\n"
```

That would work for me, because then in this PR I could invoke `black.format_file_contents` as:

```python
formatted_str = await loop.run_in_executor(
    executor, partial(black.format_file_contents, req_str, fast=fast, mode=mode, newline=newline)
)
```

Should I wait for #3257 to be merged and then apply this change? A possible patch is using `os.linesep`, but that completely ignores the current approach (Black takes the first newline in the file as a reference)

Reply:
I will hopefully be able to land #3257 soon.

Reply:
Now that #3257 has been merged, I'll rebase my PR and apply the changes listed above 👌 (Creating the `newline` variable first, pass the argument to `black.format_file_contents`, and use the variable in the code introduced by #3257 )